lib/db2.js

```diff
@@ -50,6 +51,8 @@ function DB2(settings) {
     ';PORT=' + this.portnumber +
     ';PROTOCOL=' + this.protocol;
 
+  this.connStr = this.dsn || this.connStrGenerate;
+
   this.schema = this.username;
   if (settings.schema) {
     this.connStr += ';CurrentSchema=' + settings.schema;
```

Comment:
@jannyHou @bajtos:  Janny and I discussed this in the hall.  this.schema must always be set as it is used throughout the code.  This means we will have to parse the DSN and extract username and schema. If schema is null then this.schema should be set to the username value.